codersdk/workspaceagents.go

```diff
@@ -331,21 +331,21 @@ func (c *Client) DialWorkspaceAgentTailnet(ctx context.Context, logger slog.Logg
 				// Need to disable compression to avoid a data-race.
 				CompressionMode: websocket.CompressionDisabled,
 			})
-			if err != nil {
-				if errors.Is(err, context.Canceled) {
-					return
-				}
-				logger.Debug(ctx, "failed to dial", slog.Error(err))
-				continue
-			}
 			if isFirst {
-				if res.StatusCode == http.StatusConflict {
+				if err != nil && res.StatusCode == http.StatusConflict {
```

Comment:
Should this be `err == nil`? Thinking this will still have the potential nil pointer deref if `err != nil`?

Reply:
Ahh, you're correct. I'll fix this now in another PR. I merged due to panic of `main` being busted!

Reply:
It seems the error isn't nil there actually, so a weird case. I'm just going to check for the status code, then it should be fine!

Reply:
https://github.com/coder/coder/pull/4327